front/app/containers/SpamReport/index.tsx

```diff
@@ -56,21 +57,21 @@ const SpamReportForm = ({ resourceType, resourceId }: Props) => {
       return;
     }
 
-    setLoading(true);
-
-    sendSpamReport(resourceType, resourceId, diff)
-      .then(() => {
-        setLoading(false);
-        setSaved(true);
-        setErrors(null);
-        setDiff(null);
-      })
-      .catch((e) => {
-        setLoading(false);
-        if (isCLErrorJSON(e)) {
-          setErrors(e.json.errors);
-        }
-      });
+    addSpamReport(
+      { targetId, targetType, spam_report: diff },
+      {
+        onSuccess: () => {
+          setSaved(true);
+          setErrors(null);
+          setDiff(null);
+        },
+        onError: (e) => {
+          if (isCLErrorJSON(e)) {
+            setErrors(e.json.errors);
```

Comment:
You could get the errors object directly from the mutation, instead of using a custom component state for it. Also, keep in mind that the `fetcher` works a little differently so there will no longer be `json` in the error object - it will just be `e.errors` in this case.